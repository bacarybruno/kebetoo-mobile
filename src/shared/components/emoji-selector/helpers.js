import { Platform } from 'react-native'

import categories, { fullCategoryNames as categoryNames } from './categories'

const charFromUtf16 = (utf16) => String.fromCodePoint(...utf16.split('-').map((u) => `0x${u}`))
const charFromEmojiObject = (obj) => charFromUtf16(obj.unified)

// filters
const filterSupported = (emoji) => (
  !emoji.obsoleted_by
  && emoji[`has_img_${Platform.OS === 'ios' ? 'apple' : 'google'}`]
  && emoji.category !== 'Skin Tones'
)

// sort
const sortByCategory = (a, b) => (
  categoryNames.indexOf(a.category) - categoryNames.indexOf(b.category)
)
const sortBySortOrder = (a, b) => a.sort_order - b.sort_order
const sortByCategoryAndOrder = (a, b) => {
  const categorySort = sortByCategory(a, b)
  const orderSort = sortBySortOrder(a, b)
  return categorySort || orderSort
}

// map
const mapEmojis = (emoji) => ({
  symbol: charFromEmojiObject(emoji),
  shortNames: emoji.short_names,
  category: emoji.category === 'People & Body' ? 'Smileys & Emotion' : emoji.category,
})

// group
const groupByCategory = (emojis) => {
  const groups = {}
  emojis.forEach((emoji) => {
    if (groups[emoji.category] === undefined) groups[emoji.category] = []
    groups[emoji.category].push(emoji)
  })
  if (groups[categories.history.name] === undefined) groups[categories.history.name] = []
  return groups
}

// history params
const historyStorageKey = '@app/storage/emoji-history'
const historyMaxItems = 40

export default {
  sortByCategoryAndOrder,
  charFromEmojiObject,
  filterSupported,
  groupByCategory,
  mapEmojis,
  history: {
    storageKey: historyStorageKey,
    maxItems: historyMaxItems,
  },
}
