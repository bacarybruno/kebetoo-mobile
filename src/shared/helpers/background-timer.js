import { DeviceEventEmitter, NativeAppEventEmitter, Platform } from 'react-native';
import RNBackgroundTimer from 'react-native-background-timer';

const EventEmitter = Platform.select({
  ios: NativeAppEventEmitter,
  android: DeviceEventEmitter,
});

class BackgroundTimer {
  static setInterval(callback, delay) {
    RNBackgroundTimer.start();
    this.backgroundListener = EventEmitter.addListener('backgroundTimer', () => {
      this.backgroundTimer = RNBackgroundTimer.setInterval(callback, delay);
    });
    return this.backgroundListener;
  }

  static clearInterval(timer) {
    if (timer) timer.remove();
    if (this.backgroundTimer) RNBackgroundTimer.clearInterval(this.backgroundTimer);
    RNBackgroundTimer.stop();
  }
}

export default BackgroundTimer;
