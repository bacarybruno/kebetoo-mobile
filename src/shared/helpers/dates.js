import dayjs from 'dayjs'

export const readableSeconds = (value) => {
  const date = dayjs().startOf('day').second(value)
  return date.format('mm:ss')
}