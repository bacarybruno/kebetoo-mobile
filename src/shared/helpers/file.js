export const mimeTypes = Object.freeze({
  // images
  png: 'image/png',
  jpg: 'image/jpeg',
  jpeg: 'image/jpeg',
  gif: 'image/gif',
  // audio
  mp3: 'audio/mp3',
  m4a: 'audio/mp4',
  aac: 'audio/x-aac',
  opus: 'audio/ogg',
  ogg: 'audio/ogg',
  oga: 'audio/ogg',
  // video
  mov: 'video/quicktime',
  mpg: 'video/mpeg',
  mpeg: 'video/mpeg',
  mp4: 'video/mp4',
  wmv: 'video/x-ms-wmv',
  avi: 'video/x-msvideo',
  webm: 'video/webm',
})

const videoExtensions = Object
  .keys(mimeTypes)
  .filter((mime) => mimeTypes[mime].startsWith('video'))

export const getFileName = (path) => {
  if (typeof path !== 'string') return undefined
  const parts = path.split('/')
  return parts[parts.length - 1]
}

export const getExtension = (filename) => {
  if (typeof filename !== 'string') return undefined
  const parts = filename.split('.')
  return parts[parts.length - 1]
}

export const getMimeType = (filename) => {
  const extension = getExtension(filename)
  return mimeTypes[extension]
}

export const getMediaType = (filename) => {
  const mimeType = getMimeType(filename)
  if (mimeType === undefined) return undefined
  return mimeType.split('/')[0]
}

export const isVideo = (filename) => {
  const extension = getExtension(filename)
  return videoExtensions.includes(extension)
}
