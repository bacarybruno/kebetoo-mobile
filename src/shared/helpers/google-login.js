import { GoogleSignin, statusCodes } from '@react-native-community/google-signin'
import auth from '@react-native-firebase/auth'

export default async () => {
  const result = { error: null, data: null }
  try {
    await GoogleSignin.hasPlayServices()
    await GoogleSignin.signOut()
    const { idToken } = await GoogleSignin.signIn()
    const googleCredential = auth.GoogleAuthProvider.credential(idToken)
    result.data = await auth().signInWithCredential(googleCredential)
  } catch (error) {
    if (error.code === statusCodes.SIGN_IN_CANCELLED) {
      // user cancelled the login flow
    } else if (error.code === statusCodes.IN_PROGRESS) {
      // operation (e.g. sign in) is in progress already
    } else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
      // play services not available or outdated
    } else {
      // some other error happened
    }
    result.error = error
  }
  return Promise.resolve(result)
}
