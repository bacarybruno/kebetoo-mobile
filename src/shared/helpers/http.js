/* eslint-disable no-undef */
import RNFetchBlob from 'rn-fetch-blob'

export const BASE_URL = 'http://localhost:1337'
const ITEMS_PER_PAGE = 20

const parseJSON = (res) => res.json()

const getApiUrl = (path) => `${BASE_URL}/${path}`

const request = {
  get: (path = '') => fetch(getApiUrl(path)).then(parseJSON),

  post: (path = '', body) => fetch(getApiUrl(path), {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(body),
  }).then(parseJSON),

  put: (path = '', body) => fetch(getApiUrl(path), {
    method: 'PUT',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(body),
  }).then(parseJSON),

  delete: (path = '') => fetch(getApiUrl(path), {
    method: 'DELETE',
  }).then(parseJSON),

  postFormData: (path = '', formdata) => RNFetchBlob.fetch('POST', getApiUrl(path), {
    'Content-Type': 'multipart/form-data',
  }, formdata).then(parseJSON),
}

export const getPosts = () => request.get(`posts?_limit=${ITEMS_PER_PAGE}`)

export const getUserPosts = (authorId) => request.get(
  `posts?author=${authorId}&_sort=updatedAt:desc`,
)

export const getPost = (id) => request.get(`posts/${id}`)

export const getComments = (postId) => request.get(`comments?post.id=${postId}`)

export const getLatestsPosts = (page = 0) => request.get(
  `posts?_sort=updatedAt:desc&_start=${page * ITEMS_PER_PAGE}&_limit=${ITEMS_PER_PAGE}`,
)

export const createPost = ({ author, content }) => request.post('posts', { author, content })
export const createPostWithAudio = ({ author, audio, content }) => {
  const postData = {
    name: 'data',
    data: JSON.stringify({ content, author }),
  }
  const audioData = {
    name: 'files.audio',
    filename: audio.name,
    type: audio.mimeType,
    data: RNFetchBlob.wrap(audio.uri),
  }
  return request.postFormData('posts', [postData, audioData])
}
export const createPostWithImage = ({ author, image, content }) => {
  const postData = {
    name: 'data',
    data: JSON.stringify({ content, author }),
  }
  const imageData = {
    name: 'files.image',
    filename: image.name,
    type: image.mimeType,
    data: RNFetchBlob.wrap(image.uri),
  }
  return request.postFormData('posts', [postData, imageData])
}

export const editPost = ({ id, content }) => request.put(`posts/${id}`, { content })

export const deletePost = (id) => request.delete(`posts/${id}`)

export const commentPost = ({ author, post, content }) => request.post(
  'comments',
  { author, post, content },
)
export const commentPostWithAudio = ({ author, audio, post }) => {
  const commentData = {
    name: 'data',
    data: JSON.stringify({ author, post }),
  }
  const audioData = {
    name: 'files.audio',
    filename: audio.name,
    type: audio.mimeType,
    data: RNFetchBlob.wrap(audio.uri),
  }
  return request.postFormData('comments', [commentData, audioData])
}
export const deleteComment = (id) => request.delete(`comments/${id}`)

export const createReaction = (type, post, author) => (
  request.post('reactions/', { type, post, author })
)
export const createCommentReaction = (type, comment, author) => (
  request.post('reactions/', { type, comment, author })
)
export const editReaction = (id, type) => request.put(`reactions/${id}`, { type })
export const deleteReaction = (id) => request.delete(`reactions/${id}`)

export const getPostsCount = (author) => request.get(`posts/count?author=${author}`)
export const getReactionsCount = (author) => request.get(`reactions/count?post.author=${author}`)
export const getCommentsCount = (author) => request.get(`comments/count?post.author=${author}`)
