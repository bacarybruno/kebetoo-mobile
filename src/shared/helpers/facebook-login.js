import { LoginManager, AccessToken } from 'react-native-fbsdk'
import auth from '@react-native-firebase/auth'

import { createUser } from '@app/shared/helpers/users'

export const getFacebookPicture = (profileId) => `https://graph.facebook.com/${profileId}/picture?type=large`

// TODO: handle result errors
const facebookLogin = async () => {
  const result = { error: null, data: null }
  try {
    LoginManager.logOut()
    const loginResult = await LoginManager.logInWithPermissions(['public_profile', 'email'])
    const { accessToken } = await AccessToken.getCurrentAccessToken()
    const facebookCredential = auth.FacebookAuthProvider.credential(accessToken)
    const data = await auth().signInWithCredential(facebookCredential)

    const { additionalUserInfo, user } = data
    const { profile } = additionalUserInfo
    const { id: profileId } = profile

    const photoURL = getFacebookPicture(profileId)
    await createUser({ id: user.uid, displayName: user.displayName, photoURL })

    user.updateProfile({ photoURL })

    result.data = data
  } catch (error) {
    result.error = error
  }
  return Promise.resolve(result)
}

export default facebookLogin
