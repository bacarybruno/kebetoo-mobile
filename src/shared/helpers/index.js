import * as dates from './dates';
import * as file from './file';

export { default as generateColor } from './color-generator';
export { dates, file };
