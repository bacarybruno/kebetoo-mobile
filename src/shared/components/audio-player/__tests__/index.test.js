import { fireEvent } from 'react-native-testing-library';
import Ionicon from 'react-native-vector-icons/Ionicons';
import { ActivityIndicator } from 'react-native';

import setupTest from '@app/config/jest-setup';
import { readableSeconds } from '@app/shared/helpers/dates';

import AudioPlayer, { PlayButton, DeleteIconButton } from '../index';

jest.useFakeTimers();

const givenAudioPlayer = setupTest(AudioPlayer)({
  source: 'http://localhost:1337/uploads/PTT_20200618_4_77ef44ec7b.mp3',
  style: {
    height: 20,
  },
});

it('renders AudioPlayer', () => {
  const { wrapper } = givenAudioPlayer();
  expect(wrapper.toJSON()).toMatchSnapshot();
});

it('renders AudioPlayer with delete icon', () => {
  const { wrapper } = givenAudioPlayer({
    onDelete: jest.fn(),
  });
  expect(wrapper.toJSON()).toMatchSnapshot();
});

it('renders round AudioPlayer', () => {
  const { wrapper } = givenAudioPlayer({
    round: true,
  });
  expect(wrapper.toJSON()).toMatchSnapshot();
});

describe('player states', () => {
  it('shows default state on render', () => {
    const { wrapper } = givenAudioPlayer();
    expect(wrapper.root.findByProps({ text: readableSeconds(0) })).toBeTruthy();
    expect(wrapper.root.findByProps({ testID: 'progress' }).props.style).toMatchObject({ width: '0%' });
    expect(wrapper.root.findByType(Ionicon).props.name).toBe('ios-play');
  });
  it('initialize player', () => {
    const { wrapper } = givenAudioPlayer();
    fireEvent.press(wrapper.root.findByType(PlayButton));
    wrapper.root.findByProps({ testID: 'audio-player' }).props.onLoadStart();
    expect(wrapper.root.findAllByType(ActivityIndicator).length).toBe(1);
  });
  it('play pause audio', () => {
    const { wrapper } = givenAudioPlayer();
    fireEvent.press(wrapper.root.findByType(PlayButton));
    expect(wrapper.root.findByProps({ testID: 'audio-player' }).props.paused).toBe(false);
    expect(wrapper.root.findByType(Ionicon).props.name).toBe('ios-pause');
    fireEvent.press(wrapper.root.findByType(PlayButton));
    expect(wrapper.root.findByProps({ testID: 'audio-player' }).props.paused).toBe(true);
    expect(wrapper.root.findByType(Ionicon).props.name).toBe('ios-play');
  });
  it('end playing', () => {
    const { wrapper } = givenAudioPlayer();
    fireEvent.press(wrapper.root.findByType(PlayButton));
    expect(wrapper.root.findByProps({ testID: 'audio-player' }).props.paused).toBe(false);
    expect(wrapper.root.findByType(Ionicon).props.name).toBe('ios-pause');
    wrapper.root.findByProps({ testID: 'audio-player' }).props.onEnd();
    expect(wrapper.root.findByProps({ text: readableSeconds(0) })).toBeTruthy();
    expect(wrapper.root.findByProps({ testID: 'progress' }).props.style).toMatchObject({ width: '0%' });
  });
});

describe('player wrapper', () => {
  it('handles player press', () => {
    const { wrapper, props } = givenAudioPlayer({
      onPress: jest.fn(),
    });
    const playerWrapper = wrapper.root.findByProps({ testID: 'player-wrapper' });
    playerWrapper.props.onPress();
    expect(props.onPress).toBeCalledTimes(1);
  });
  it('toggle player if no custom onPress handler', () => {
    const { wrapper } = givenAudioPlayer();
    fireEvent.press(wrapper.root.findByProps({ testID: 'player-wrapper' }));
    expect(wrapper.root.findByProps({ testID: 'audio-player' }).props.paused).toBe(false);
  });
});
it('handles delete button', () => {
  const { wrapper, props } = givenAudioPlayer({
    onDelete: jest.fn(),
  });
  const deleteButton = wrapper.root.findByType(DeleteIconButton);
  expect(deleteButton).toBeTruthy();
  fireEvent.press(deleteButton);
  expect(props.onDelete).toBeCalledTimes(1);
});
