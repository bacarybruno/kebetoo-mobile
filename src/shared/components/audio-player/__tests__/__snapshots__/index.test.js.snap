// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AudioPlayer 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 52,
        "maxHeight": 52,
      },
      Object {
        "height": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "borderRadius": 5,
          "flex": 1,
          "overflow": "hidden",
        },
        Object {},
        undefined,
      ]
    }
    testID="player-wrapper"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(199, 199, 204, 255)",
            "borderBottomLeftRadius": 5,
            "borderTopLeftRadius": 5,
            "height": "100%",
            "opacity": 0.4,
            "position": "absolute",
            "width": "0%",
          }
        }
        testID="progress"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderRadius": 15,
            "elevation": 2,
            "height": 30,
            "justifyContent": "center",
            "marginLeft": 16,
            "opacity": 1,
            "shadowColor": "rgba(0, 0, 0, 255)",
            "shadowOffset": Object {
              "x": 0,
              "y": 1,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 1.41,
            "width": 30,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "#0B1B77",
              "fontFamily": "Ionicons",
              "fontSize": 22,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
          text="ionicon-ios-play"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/waves.png",
            }
          }
          style={
            Object {
              "resizeMode": "contain",
              "width": "80%",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            false,
            false,
            false,
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 16,
            },
            Array [
              Object {
                "color": "rgba(0,0,0,0.54)",
              },
              Object {
                "backgroundColor": "transparent",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "bottom": 2,
                "position": "absolute",
                "right": 8,
              },
            ],
          ]
        }
      >
        0:00
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders AudioPlayer with delete icon 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 52,
        "maxHeight": 52,
      },
      Object {
        "height": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "borderRadius": 5,
          "flex": 1,
          "overflow": "hidden",
        },
        Object {},
        undefined,
      ]
    }
    testID="player-wrapper"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(199, 199, 204, 255)",
            "borderBottomLeftRadius": 5,
            "borderTopLeftRadius": 5,
            "height": "100%",
            "opacity": 0.4,
            "position": "absolute",
            "width": "0%",
          }
        }
        testID="progress"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderRadius": 15,
            "elevation": 2,
            "height": 30,
            "justifyContent": "center",
            "marginLeft": 16,
            "opacity": 1,
            "shadowColor": "rgba(0, 0, 0, 255)",
            "shadowOffset": Object {
              "x": 0,
              "y": 1,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 1.41,
            "width": 30,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "#0B1B77",
              "fontFamily": "Ionicons",
              "fontSize": 22,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
          text="ionicon-ios-play"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/waves.png",
            }
          }
          style={
            Object {
              "resizeMode": "contain",
              "width": "80%",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            false,
            false,
            false,
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 16,
            },
            Array [
              Object {
                "color": "rgba(0,0,0,0.54)",
              },
              Object {
                "backgroundColor": "transparent",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "bottom": 2,
                "position": "absolute",
                "right": 8,
              },
            ],
          ]
        }
      >
        0:00
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 50,
        "left": 50,
        "right": 50,
        "top": 50,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "borderRadius": 15,
        "elevation": 2,
        "height": 20,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": -5,
        "shadowColor": "rgba(0, 0, 0, 255)",
        "shadowOffset": Object {
          "x": 0,
          "y": 1,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 1.41,
        "top": -5,
        "width": 20,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Object {
          "color": "rgba(0,0,0,0.87)",
          "fontFamily": "Ionicons",
          "fontSize": 20,
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
      text="ionicon-ios-close"
    />
  </View>
</View>
`;

exports[`renders round AudioPlayer 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 52,
        "maxHeight": 52,
      },
      Object {
        "height": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "borderRadius": 5,
          "flex": 1,
          "overflow": "hidden",
        },
        Object {},
        Object {
          "borderRadius": 22,
        },
      ]
    }
    testID="player-wrapper"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(199, 199, 204, 255)",
            "borderBottomLeftRadius": 5,
            "borderTopLeftRadius": 5,
            "height": "100%",
            "opacity": 0.4,
            "position": "absolute",
            "width": "0%",
          }
        }
        testID="progress"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderRadius": 15,
            "elevation": 2,
            "height": 30,
            "justifyContent": "center",
            "marginLeft": 16,
            "opacity": 1,
            "shadowColor": "rgba(0, 0, 0, 255)",
            "shadowOffset": Object {
              "x": 0,
              "y": 1,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 1.41,
            "width": 30,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "#0B1B77",
              "fontFamily": "Ionicons",
              "fontSize": 22,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
          text="ionicon-ios-play"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/waves.png",
            }
          }
          style={
            Object {
              "resizeMode": "contain",
              "width": "80%",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            false,
            false,
            false,
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 16,
            },
            Array [
              Object {
                "color": "rgba(0,0,0,0.54)",
              },
              Object {
                "backgroundColor": "transparent",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "bottom": 2,
                "position": "absolute",
                "right": 8,
              },
            ],
          ]
        }
      >
        0:00
      </Text>
    </View>
  </View>
</View>
`;
